File: backend/src/main/java/com/hmdp/entity/Voucher.java
package com.hmdp.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.time.LocalDateTime;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("tb_voucher")
public class Voucher extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long shopId;
    private String title;
    private String subTitle;
    private Long payValue;
    private Long actualValue;
    private Integer stock;
    private LocalDateTime beginTime;
    private LocalDateTime endTime;
    private Integer status;
}

File: frontend/src/api/voucher.ts
import { http } from './http'

export interface Voucher {
  id: number
  shopId: number
  title: string
  subTitle: string
  payValue: number
  actualValue: number
  stock: number
  beginTime: string
  endTime: string
  status: number
}

export async function fetchSeckillVouchers() {
  const { data } = await http.get('/vouchers/seckill')
  return data.data as Voucher[]
}

export async function claimSeckillVoucher(voucherId: number) {
  const { data } = await http.post(`/vouchers/seckill/${voucherId}/claim`)
  return data.data as number
}
